src/channels.rs

```diff
@@ -60,7 +60,7 @@ impl MiEI {
             .filter(|x| self.role_exists(x))
             .map(|x| x.to_string())
             .collect::<Vec<String>>();
-        let created_roles = new_roles
+        let _created_roles = new_roles
```

Comment:
Why create a variable at all? I don't understand what this function is for :/ maybe some documentation is in order ;)

Reply:
Also, above you are cloning a bunch of strings. Remember `to_string`, `to_owned` and `clone` are all the same when applied to strings. You could just remove the strings that don't match from the passed vector instead of creating a new one.

Alternatively you could also just iterate/filter/map directly

Reply:
Also, above you are cloning a bunch of strings. Remember to_string, to_owned and clone are all the same when applied to strings. You could just remove the strings that don't match from the passed vector instead of creating a new one.

Alternatively you could also just iterate/filter/map directly

Reply:
That function is related to another issue, it has no use right now, I can remove it if you want